docs/DeveloperGuide.md

```diff
@@ -284,38 +283,79 @@ file choosers, pop-up windows) will be similar.
 
-* For import, the `Model` component is accessed to set the new AddressBook of contacts. On the other hand, export filters the AddressBook of the `Model` using the tags specified in the user command to retrieve contacts to be exported.
+* For import, the `Model` component is accessed to set the new AddressBook of contacts. For merge, contacts are merged with the current AddressBook. On the other hand, export filters the AddressBook of the `Model` using the tags specified in the user command to retrieve contacts to be exported.
 
 * Functions in `Storage` were used to write AddressBooks to JSON files as well as read and convert JSON files to AddressBook objects.
 
-The following sequence diagram shows how the `import` operation works:
+#### Import Implementation
+
+The `ImportCommand` allows users to import contact files to the Teletubbies app. The following sequence diagram shows how the `import` operation works:
 
-![](images/ImportSequenceDiagram.png)
+<img src="images/ImportSequenceDiagram.png" width="750" />
 
 <div markdown="span" class="alert alert-info">:information_source: **Note:** The lifeline for `ImportCommand` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
 </div>
 
+* After an `ImportCommand` is created, it will be executed by the `LogicManager`. During the execution, a new `CommandResult` is returned with the `ImportUiConsumer`.
+* The `UiEffect` of the `CommandResult` is executed by the `MainWindow`. The `ImportUiConsumer` then calls `MainWindow#handleImport`, which opens the file chooser for the user to choose a file to import. 
+* When the file is selected, the `ImportUiConsumer` converts the JSON file to an AddressBook and calls `Model#setAddressBook`. The updated contact list is then displayed in the GUI.
+
+#### Merge Implementation
+
+Merge functions in a similar way to import. However, instead of replacing the AddressBook, the incoming Persons are merged with the prevailing AddressBook.
+
+Teletubbies provides commands for users to modify contacts by editing their particulars or tagging them. Since a Person can change drastically, each Person is issued a Universally Unique Identifier (UUID) to facilitate the merging process. If there is a match for the UUID in the AddressBook, the incoming Person would replace it. Else, the Person is new and would be added to the AddressBook.
+
+The following sequence diagram shows how the `merge` operation works:
+
+<img src="images/MergeSequenceDiagram.png" width="750" />
+
+<div markdown="span" class="alert alert-info">:information_source: **Note:** The lifeline for `MergeCommand` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+</div>
+
+#### Export Implementation
+
+Export is distinct from other features as it comprises of 2 commands, as illustrated in this activity diagram:
+
+![ExportActivityDiagram](images/ExportActivityDiagram.png)
```

Comment:
I'm really not sure about excluding the implementation details regarding export and confirmExport's interaction with Model, but i can add another diagram to explain those 2 methods specifically? what do you think?

Reply:
So maybe like another sequence diagram?

Reply:
maybe something like this with the ref?

![image](https://user-images.githubusercontent.com/68676914/140675534-6256d10b-f29c-4c80-892f-fdf530513b39.png)